analyses/snv-callers/util/plot_functions.R

```diff
@@ -0,0 +1,274 @@
+  #   vaf_df: Need a MAF data.frame that has `vaf`, `base_change` and
+  #           `experimental_strategy` columns that are added with the
+              `set_up_maf` function.
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #   `both`.
+  #   cosmic_clean_file: a file path to a TSV of COSMIC mutations that has been
+  #                     been cleaned up to have the genomic coordinates separated
+  #                     into Chr, Start, and End columns. This is passed to the
+  #                     `find_cosmic_overlap` function.
+  #
+  # Returns:
+  #  A violin plot with VAF plotted by whether or not it is overlapping with the
+  #  COSMIC mutation set.
+  #
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy if specified
+  if (exp_strategy != "BOTH") {
+    vaf_df <- vaf_df %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Get the overlap with COSMIC mutations using special function
+  maf_cosmic <- find_cosmic_overlap(vaf_df, cosmic_clean_file)
+
+  # Plot this as a violin plot
+  barplot <- maf_cosmic %>%
+    ggplot2::ggplot(ggplot2::aes(x = overlap_cosmic, y = vaf))
+
+  # Get rid of legend if both data aren't being plotted
+  if (exp_strategy != "BOTH") {
+    barplot <- barplot +
+      ggplot2::geom_violin(fill = "navyblue") +
+      ggplot2::theme(legend.position = "none")
+  } else {
+    barplot <- barplot +
+      ggplot2::geom_violin(ggplot2::aes(fill = experimental_strategy))
+  }
+  # Add some aesthetics
+  barplot <- barplot +
+    ggplot2::theme_classic() +
+    ggplot2::xlab("Same mutation and base change found in COSMIC") +
+    colorblindr::scale_fill_OkabeIto()
+
+  return(barplot)
+}
+
+tmb_plot <- function(tmb_df, exp_strategy = "BOTH", x_axis = "short_histology") {
+  # Plot Tumor Mutational Burden as a jitter plot by a variable of choice.
+  #
+  # Args:
+  #   tmb_df: MAF formatted data that has been turned into a data.frame and has
+  #           been run through `vaf`
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #   `both`.
+  #   x_axis: what variable you would like to use to plot on the x-axis. Default
+  #           is `short_histology` column.
+  #
+  # Returns:
+  # A jitterplot that plots the TMB stats by the argument specified in x_axis
+  # argument.
+  #
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy ifspecified
+  if (exp_strategy != "BOTH") {
+    tmb_df <- tmb_df %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Plot this as a jitter plot
+  barplot <- tmb_df %>%
+    ggplot2::ggplot(ggplot2::aes(x = eval(parse(text = x_axis)), y = tmb))
+
+  # Get rid of legend if both data aren't being plotted
+  if (exp_strategy != "BOTH") {
+    barplot <- barplot +
+      ggplot2::geom_jitter(color = "navyblue", alpha = 0.2) +
```

Comment:
I don't like jitter plots (ask me why!) maybe a ggforce::geom_sina?

Reply:
The main reason this is a jitterplot is because of the [plot that it was requested to be based on in the issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/3#issuecomment-516395431) I can play around with a sina plot though and see what that looks like.